File: backup_before_industrialisation_20250626/backend/components/metiers/video/fixtures/core/validators/fixtures_validator.js
// fixtures_validator.js - Validation avancée des fixtures Video

module.exports = {
  isValidvideoModel: (model) => {
    return model && typeof model.id === 'string' && Array.isArray(model.vertices) && Array.isArray(model.faces);
  },
  isValidUser: (user) => {
    return user && typeof user.id === 'string' && typeof user.role === 'string';
  }
};

File: backend/components/metiers/industrie/fixtures/core/validators/fixtures_validator.py
"""
fixtures_validator.py - Validation avancée des fixtures Industrie (Python)
"""


def is_valid_industrie_model(model):
    """Validate industrie model fixture.

    Args:
        model (dict): The industrie model fixture to validate.

    Returns:
        bool: True if the fixture is valid, False otherwise.
    """
    return bool(
        model
        and isinstance(model.get("id"), str)
        and isinstance(model.get("vertices"), list)
        and isinstance(model.get("faces"), int)
    )


def is_valid_user(user):
    """Validate user fixture.

    Args:
        user (dict): The user fixture to validate.

    Returns:
        bool: True if the fixture is valid, False otherwise.
    """
    return bool(
        user
        and isinstance(user.get("id"), str)
        and isinstance(user.get("role"), str)
    )
